projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java

```diff
@@ -72,19 +76,23 @@ public void updateJobSecrets(String jobName, Map<String, Object> secrets) {
     String updatedSecretsString = new JSONObject(updatedSecrets).toString();
     if (updatedSecretsString.getBytes(StandardCharsets.UTF_8).length > sizeLimitBytes) {
       throw new DataJobSecretsSizeLimitException(
-          jobName, "Secret size exceeds configured limit of:" + sizeLimitBytes);
+          secretKey, "Secret size exceeds configured limit of:" + sizeLimitBytes);
     }
 
     vaultJobSecrets.setSecretsJson(updatedSecretsString);
 
-    vaultOperations.opsForVersionedKeyValue(SECRET).put(jobName, vaultJobSecrets);
+    vaultOperations.opsForVersionedKeyValue(SECRET).put(secretKey, vaultJobSecrets);
   }
 
   @Override
-  public Map<String, Object> readJobSecrets(String jobName) throws JsonProcessingException {
-    checkJobName(jobName);
+  public Map<String, Object> readJobSecrets(String teamName, String jobName)
+      throws JsonProcessingException {
+    checkInputs(teamName, jobName);
+
+    String secretKey = teamName + "/" + jobName;
```

Comment:
Extract the secret ket building in a separate method.

Reply:
fixed